slash/react/src/Popover/PopoverClick.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { PopoverProps } from "./Popover.types";
+import { PopoverBase } from "./PopoverBase";
+
+export const PopoverClick = ({
+  children,
+  placement,
+  className,
+  classModifier,
+  element,
+}: PopoverProps) => {
+  const wrapperRef = React.useRef(null);
+  const [isOpen, setOpen] = React.useState(false);
+  const [isPopoverHover, setPopoverHover] = React.useState(false);
+
+  const handleClick = (
+    event:
+      | MouseEvent
+      | React.MouseEvent
+      | React.KeyboardEvent<HTMLDivElement>
+      | React.FocusEvent,
+    isOpenValue?: boolean,
+  ) => {
+    if (isPopoverHover || !wrapperRef.current || isOpenValue === isOpen) {
+      event.stopPropagation();
+      return;
+    }
+
+    setOpen((oldIsOpen) => isOpenValue ?? !oldIsOpen);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === "Enter") {
+      handleClick(event);
+    }
+  };
+
+  const handleMouseEnterPopover = () => {
+    setPopoverHover(true);
+  };
+
+  const handleMouseLeavePopover = () => {
+    setPopoverHover(false);
+  };
```

Comment:
les fonctions sont à mettre dans des useCallback affin d'éviter de le redéfinir à chaque render

Reply:
Je doute fort que ca soit une bonne idée. Si on se réfère a la [bible sur quand utiliser les useCallback](https://kentcdodds.com/blog/usememo-and-usecallback) on est pas dans un cas utile. Donc je vais laisser tel quel pour éviter de rendre les choses pires